src/Compilers/CSharp/Test/Semantic/Semantics/NonTrailingNamedArgumentsTests.cs

```diff
@@ -0,0 +1,418 @@
+                //         M(c: 1, 2);
+                Diagnostic(ErrorCode.ERR_NoCorrespondingArgument, "M").WithArguments("a", "C.M(int, int, int)").WithLocation(10, 9)
+                );
+
+            var tree = comp.SyntaxTrees.First();
+            var model = comp.GetSemanticModel(tree);
+            var nodes = tree.GetCompilationUnitRoot().DescendantNodes();
+            var invocation = nodes.OfType<InvocationExpressionSyntax>().ElementAt(1);
+            Assert.Equal("M(c: 1, 2)", invocation.ToString());
+            AssertEx.Equal(new[] { "void C.M(System.Int32 a, System.Int32 b, [System.Int32 c = 1])" },
+                model.GetSymbolInfo(invocation).CandidateSymbols.Select(c => c.ToTestDisplayString()));
+        }
+
+        [Fact]
+        public void TestNamedParams()
+        {
+            var source = @"
+class C
+{
+    static void M(params int[] x)
+    {
+    }
+    static void Main()
+    {
+        M(x: 1, 2);
+    }
+}";
+            var comp = CreateStandardCompilation(source, parseOptions: TestOptions.Regular7_2);
+            comp.VerifyDiagnostics(
+                // (9,11): error CS8321: Named argument 'x' is used out-of-position but is followed by an unnamed argument
+                //         M(x: 1, 2);
+                Diagnostic(ErrorCode.ERR_BadNonTrailingNamedArgument, "x").WithArguments("x").WithLocation(9, 11)
+                );
+
+            var tree = comp.SyntaxTrees.First();
+            var model = comp.GetSemanticModel(tree);
+            var nodes = tree.GetCompilationUnitRoot().DescendantNodes();
+            var invocation = nodes.OfType<InvocationExpressionSyntax>().Single();
+            Assert.Equal("M(x: 1, 2)", invocation.ToString());
+            Assert.Null(model.GetSymbolInfo(invocation).Symbol); // PROTOTYPE(non-trailing)
+        }
+
+        [Fact]
+        public void TestNamedParams2()
+        {
+            var source = @"
+class C
+{
+    static void M(params int[] x)
+    {
+    }
+    static void Main()
+    {
+        M(x: 1, x: 2);
+    }
+}";
+            var comp = CreateStandardCompilation(source, parseOptions: TestOptions.Regular7_2);
+            comp.VerifyDiagnostics(
+                // (9,17): error CS1740: Named argument 'x' cannot be specified multiple times
+                //         M(x: 1, x: 2);
+                Diagnostic(ErrorCode.ERR_DuplicateNamedArgument, "x").WithArguments("x").WithLocation(9, 17)
+                );
+
+            var tree = comp.SyntaxTrees.First();
+            var model = comp.GetSemanticModel(tree);
+            var nodes = tree.GetCompilationUnitRoot().DescendantNodes();
+            var invocation = nodes.OfType<InvocationExpressionSyntax>().Single();
+            Assert.Equal("M(x: 1, x: 2)", invocation.ToString());
+            Assert.Equal("void C.M(params System.Int32[] x)", model.GetSymbolInfo(invocation).Symbol.ToTestDisplayString());
```

Comment:
Confirmed this is an existing behavior.
Filled https://github.com/dotnet/roslyn/issues/20280